marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/OutputEndpointImpl.java

```diff
@@ -61,7 +60,16 @@ private BulkOutputCallerImpl(OutputEndpointImpl endpoint) {
 
         @Override
         public void awaitCompletion() {
+            if(outputConsumer == null)
+                return;
```

Comment:
Should it throw an exception instead?  If the application hasn't supplied a consumer, is it meaningful to await completion?